From a397393ceacc0aa13a30d34be4c28b27cf8d4220 Mon Sep 17 00:00:00 2001
From: Ulya Trafimovich <skvadrik@google.com>
Date: Thu, 8 Apr 2021 14:09:31 +0100
Subject: [PATCH] dex2oat_vdex_test: add missing dependency on core boot image.

Bug: 178055795

Test: lunch armv8 \
  && rm out/host/linux-x86/framework/x86_64/core.art \
  ; m SOONG_ALLOW_MISSING_DEPENDENCIES=true \
    test-art-host-gtest-dex2oat_vdex_test64
  # failed before this CL, now passes

Change-Id: I2bc5dde0916393842cd444d256022c9b1c36343b
Merged-In: I5b104329af899ea62dc1779800264bad6b5424a4
(cherry picked from commit c60638a44e3d0b78b704665650d58044cf0195e3)
---
 build/Android.gtest.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/build/Android.gtest.mk b/build/Android.gtest.mk
index 95d0f38b67..1b2180e508 100644
--- a/build/Android.gtest.mk
+++ b/build/Android.gtest.mk
@@ -261,6 +261,10 @@ ART_GTEST_two_runtimes_test_TARGET_DEPS := $(TARGET_CORE_IMAGE_DEFAULT_64) $(TAR
 ART_GTEST_transaction_test_HOST_DEPS := $(HOST_CORE_IMAGE_DEFAULT_64) $(HOST_CORE_IMAGE_DEFAULT_32)
 ART_GTEST_transaction_test_TARGET_DEPS := $(TARGET_CORE_IMAGE_DEFAULT_64) $(TARGET_CORE_IMAGE_DEFAULT_32)

+# The dex2oat_vdex_test test has dependencies on core.oat.
+ART_GTEST_dex2oat_vdex_test_HOST_DEPS := $(HOST_CORE_IMAGE_DEFAULT_64) $(HOST_CORE_IMAGE_DEFAULT_32)
+ART_GTEST_dex2oat_vdex_test_TARGET_DEPS := $(TARGET_CORE_IMAGE_DEFAULT_64) $(TARGET_CORE_IMAGE_DEFAULT_32)
+
 ART_GTEST_dex2oat_environment_tests_HOST_DEPS := \
   $(HOST_CORE_IMAGE_optimizing_64) \
   $(HOST_CORE_IMAGE_optimizing_32) \
--
2.29.0
